hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/bucket/FileIOEngine.java

```diff
@@ -68,15 +97,18 @@ public FileIOEngine(long capacity, String... filePaths) throws IOException {
           // The next setting length will throw exception,logging this message
           // is just used for the detail reason of exception，
           String msg = "Only " + StringUtils.byteDesc(totalSpace)
-              + " total space under " + filePath + ", not enough for requested "
-              + StringUtils.byteDesc(sizePerFile);
+            + " total space under " + filePath + ", not enough for requested "
+            + StringUtils.byteDesc(sizePerFile);
           LOG.warn(msg);
         }
-        rafs[i].setLength(sizePerFile);
+        File file = new File(filePath);
+        if (file.length() != sizePerFile) {
+          rafs[i].setLength(sizePerFile);
+        }
```

Comment:
why this change?

Reply:
setLength() method will change file's last modified time. So if don't change this, the wrong time will be used to calculate checksum.

Reply:
Please add a comment block `//` to clarify the purpose.

Reply:
Ya some fat comments here would be nice.  Got why u have this check now..  I can think of a case though.  Say we have a file based cache with one file and size was 10 GB.  Now the restart of the RS happening. The cache is persisted also.  Before restart the size is been increased to 20 GB. There is no truncate and ideally the cache get rebuilt.  Only thing is after the restart the cache capacity is increased.  But now as per the code, the length is changed here and so the last modified time and which will fail the verify phase.  Is it some thing to be considered?  Dont want much complex handling for this.  Might not be a common case for persisted cache.  Max what happening is we not able to retrieve persisted cache.  But welcoming thinking/suggestion.

Reply:
If I understand you correctly, you mean change the bucket cache size before restart RS. Actually, in `retrieveFromFile()` method will check the bucket cache size: 
`if (capacitySize != cacheCapacity)
        throw new IOException("Mismatched cache capacity:"
            + StringUtils.byteDesc(capacitySize) + ", expected: "
            + StringUtils.byteDesc(cacheCapacity));`
So change the bucket cache size will no longer retrieve from file.

Reply:
Yaya that was my Q..  Oh great.. I did not read whole the code.. Then its perfect. Pls add enough comments around to know why this check and set length is so imp.  Great.